GetandTake/Services/Abstracts/IProductService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using GetandTake.Models.Base;
+using GetandTake.Models.DTOs.BaseDTO;
+using GetandTake.Models.DTOs.ListDTO;
+
+namespace GetandTake.Services.Abstracts
+{
+    public interface IProductService
+    {
+        Task<IEnumerable<ProductsDTO>> GetAllAsync();
+        Task<IEnumerable<ProductsDTO>> GetAllByCategoryAsync(int categoryId);
```

Comment:
To mark it's method concrete purpose it's better to add `CategoryId` to method name, so it will look like `GetAllByCategoryIdAsync`